### PR TITLE
Avoid the use of ArrayView for Tensors in MPI::sum().

### PR DESCRIPTION
**Pull request by bangerth**
This is the second place where we use `ArrayView` for `Tensor` objects (the other one is fixed in #12684). As discussed in #12624, we want to avoid this.

/rebuild

### CODE REVIEW

File: include/deal.II/base/mpi.templates.h [hunk 1]
```diff
@@ -323,12 +323,20 @@ namespace Utilities
...
     Tensor<rank, dim, Number>
-    sum(const Tensor<rank, dim, Number> &local,
-        const MPI_Comm &                 mpi_communicator)
+    sum(const Tensor<rank, dim, Number> &t, const MPI_Comm &mpi_communicator)
     {
-      Tensor<rank, dim, Number> sums;
-      sum(local, mpi_communicator, sums);
-      return sums;
+      // Copy the tensor into a C-style array with which we can then
+      // the other sum() function
```
Comment (masterleinad) [inline include/deal.II/base/mpi.templates.h:329]:
```suggestion
      // Copy the tensor into a C-style array with which we can then
      // call the other sum() function.
```

File: include/deal.II/base/mpi.templates.h [hunk 2]
```diff
@@ -338,6 +346,8 @@ namespace Utilities
     sum(const SymmetricTensor<rank, dim, Number> &local,
         const MPI_Comm &                          mpi_communicator)
     {
+      // Copy the tensor into a C-style array with which we can then
+      // the other sum() function
```
Comment (masterleinad) [inline include/deal.II/base/mpi.templates.h:350]:
```suggestion
      // Copy the tensor into a C-style array with which we can then
      // call the other sum() function.
```

**Review by masterleinad — COMMENTED** [review]
Otherwise, this looks good to me as well.

**Comment by bangerth** [comment]
Thanks, @masterleinad -- fixed!

**Review by masterleinad — APPROVED** [review]
Looks good to me but we should probably discuss @drwells proposal to add a `Tensor::to_array` member function that would avoid the boilerplate code here.

**Comment by bangerth** [comment]
How about we merge this now and then do what you suggest as a follow-up? It doesn't become easier/less work for anyone when we try to interleave patches from different people :-)

**Comment by masterleinad** [comment]
> How about we merge this now and then do what you suggest as a follow-up? It doesn't become easier/less work for anyone when we try to interleave patches from different people :-)

I agree, we should just not forget to come back to this place.

**Review by kronbichler — APPROVED** [review]

**Pull request merged**